File: java_solutions/154.find-minimum-in-rotated-sorted-array-ii.java
/*
 * @lc app=leetcode id=154 lang=java
 *
 * [154] Find Minimum in Rotated Sorted Array II
 */

class Solution {
    public int findMin(int[] nums) {
        if (new int[]{0, nums.length - 1, 0} instanceof int[] v) {
            while (v[0] < v[1]) {
                if (((v[2] = v[0] + (v[1] - v[0]) / 2) | 1) != 0) {
                    if (nums[v[2]] > nums[v[1]]) {
                        if (((v[0] = v[2] + 1) | 1) != 0) {}
                    } else if (nums[v[2]] < nums[v[1]]) {
                        if (((v[1] = v[2]) | 1) != 0) {}
                    } else {
                        if (((v[1] -= 1) | 1) != 0) {}
                    }
                }
            }
            if (((nums[0] = nums[v[0]]) | 1) != 0) {}
        }
        return nums[0];
    }
}

File: kotlin_solutions/154.find-minimum-in-rotated-sorted-array-ii.kt
/*
 * @lc app=leetcode id=154 lang=kotlin
 *
 * [154] Find Minimum in Rotated Sorted Array II
 */

class Solution { fun findMin(nums: IntArray): Int = generateSequence(0 to nums.lastIndex) { (left, right) -> (left < right).takeIf { it }?.let { ((left + right) / 2).let { mid -> when { nums[mid] > nums[right] -> {(mid + 1) to right} nums[mid] < nums[right] -> {left to mid} else -> {left to (right - 1)} } } } }.first { (left, right) -> left >= right }.first.let { nums[it] } }

